qiskit/providers/models/pulsedefaults.py

```diff
@@ -277,6 +278,22 @@ def from_dict(cls, data):
             in_data["discriminator"] = Discriminator.from_dict(in_data.pop("discriminator"))
         return cls(**in_data)
 
+    @property
+    def frames(self) -> FramesConfiguration:
+        """Get the frames supported by the backend.
+
+        Returns:
+            frames: A list of dicts, each dict defines a frame, its frequency, and its purpose.
+        """
+        frames = {}
```

Comment:
`sample_duration` is not configured here. This cause a problem of resolving frame since None type object cannot be operated.

Reply:
Yes this is an issue I am aware of. The problem we face is that the frequencies are in the defaults and the `sample_duration` is in the configuration. I.e. you cannot get the `sample_duration` from the defaults. The `sample_duration` is only relevant when resolving the frames. Therefore the assemble adds it to the `FramesConfiguration`, see here: https://github.com/eggerdj/qiskit-terra/blob/ee161ee6241a4e3ac4be869b18038c6465a56fa7/qiskit/compiler/assembler.py#L471-L472. The user could also do this now with:
```python
backend.defaults().frames.add_dt(backend.configuration().dt)
```
if he wants to resolve frames himself.

Reply:
This is also solved by this: 990eec5 . I.e. restricting to hardware in which all frames have the same sample duration.